File: solutions/javascript/darts/1/darts.js
export const score = (x, y) => {
  let radius = (( x ** 2) + ( y ** 2)) ** (0.5);  
  if (radius <= 1) {
    return 10;
  } else if (radius > 1 && radius <= 5) {
    return 5;
  } else if (radius > 5 && radius <= 10) {
    return 1;    
  } else {
    return 0;
  }
}

File: solutions/ruby/darts/1/darts.rb
class Darts
  def initialize(x, y)
    self.x = x
    self.y = y
    self.radius = Math.sqrt(x ** 2 + y ** 2)
  end

  attr_accessor :x, :y, :radius

  public

  def score
    case radius
    when 0..1 then 10
    when 1..5 then 5
    when 5..10 then 1
    else 0
    end
  end
end
